### ISSUE TITLE
Where did you get information for this protocol?

### ISSUE THREAD

**Issue by veegee** [mrene/node-minidsp#4]
Hi, I'd like to extend the capabilities of this library in various ways to do most things that the official software can (such as set attenuation of specific channels, mute specific channels, etc.). Where did you get the information for this protocol? Did you reverse engineer it yourself?

**Comment by mrene** [comment]
Hi!

I mostly used frida to reverse engineer it at the HID level, but then I noticed the commands were all sent over TCP so it made more sense to look at it through wireshark.

I took a quick look at the flash bytecode to make sense of some messages too. I didn't find any documentation anywhere but the official utility reads some XML configuration for each model that seem to indicate the address of every filters, etc.

**Comment by veegee** [comment]
Perfect, thank you! The XML config should be useful. If not, should be pretty straightforward to listen in on the TCP messages sent to the proxy program it launches.

**Comment by mrene** [comment]
Exactly, you can also use the proxy mode provided by node-minidsp directly (set DEBUG=* to enable a lot of debug output). As long as you bind to 5333, it won't launch the proxy and connect to it directly.
